client/src/Login.tsx

```diff
@@ -0,0 +1,165 @@
+import * as React from 'react';
+import { Link as RouterLink } from 'react-router-dom';
+import Paper from '@material-ui/core/Paper';
+import Typography from '@material-ui/core/Typography';
+import Grid from '@material-ui/core/Grid';
+import Button from '@material-ui/core/Button';
+import FormControl from '@material-ui/core/FormControl';
+import InputLabel from '@material-ui/core/InputLabel';
+import FilledInput from '@material-ui/core/FilledInput';
+import Link from '@material-ui/core/Link';
+import TextField from '@material-ui/core/TextField';
+import FacebookIcon from '@material-ui/icons/Facebook';
+import Visibility from '@material-ui/icons/Visibility';
+import VisibilityOff from '@material-ui/icons/VisibilityOff';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import IconButton from '@material-ui/core/IconButton';
+import { makeStyles, Theme } from '@material-ui/core/styles';
+
+const useStyles = makeStyles((theme: Theme) => {
+    const xPadding = 6;
+    const yPadding = 16;
+
+    return {
+        paper: {
+            width: 821,
+            height: 732,
+            borderRadion: 10,
+            paddingTop: theme.spacing(xPadding),
+            paddingBottom: theme.spacing(xPadding),
+            paddingLeft: theme.spacing(yPadding),
+            paddingRight: theme.spacing(yPadding)
+        },
+        button: {
+            borderRadius: 0,
+            height: 62
+        },
+        link: {
+            color: 'black'
+        }
+    };
+});
+
+function Login() {
+    const classes = useStyles();
+
+    interface UserLoginData {
+        email: string;
+        password: string;
+    }
+
+    const initialFormData: UserLoginData = {
+        email: '',
+        password: ''
+    };
```

Comment:
i would move `UserLoginData` and `initialFormData` out of the FC so they aren't redefined on every rerender
also, IMO `formData` should be broken up into separate pieces of state, because as it is now, the password component would rerender if the email input is changed, but that can be another PR

Reply:
Agreed, and I have done part of those changes in the Login component I have locally. Figured TypeScript conversion was higher priority so I put that one on hold. I'll have that PR up soon after this.